components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -1164,6 +1164,9 @@ def __init__(self, argument, index, args, argc, descriptorProvider,
 
         if not argument.variadic:
             if argument.optional:
+                # Treat undefined arguments as missing
+                undefined_condition = string.Template("!${args}.get(${index}).is_undefined()").substitute(replacer)
```

Comment:
`.get()` returns an undefined value when it is called on a missing argument, whereas `.index()` will fail because of the `assert!`. I guess I'll stick with `.get()` here.

Which means I'll have to remove the previous definition of `condition` up there, since it's doing double-checking of parameters.